src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -74,42 +80,69 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             if (declarationContext is VariableDeclarationSyntax varDecl)
             {
                 typeSyntax = varDecl.Type;
+                declarationSyntax = varDecl.Variables.Single().Identifier.Parent;
```

Comment:
.Single is still concerning to me. can you at least comment in the code what method will have ensured this?

Reply:
CS0819 prevents multiple declarators for implicit declarations. We don't offer (and shouldn't) to use implicit types in those cases.